src/postgres/tree/PostgresDatabaseTreeItem.ts

```diff
@@ -94,10 +99,52 @@ export class PostgresDatabaseTreeItem extends AzureParentTreeItem<ISubscriptionC
         credentialsTreeItem.commandArgs = [this.parent];
         return [credentialsTreeItem];
     }
+
     public async deleteTreeItemImpl(): Promise<void> {
         const client: PostgreSQLManagementClient = createAzureClient(this.root, PostgreSQLManagementClient);
         await client.databases.deleteMethod(azureUtils.getResourceGroupFromId(this.fullId), this.parent.name, this.databaseName);
     }
+
+    private async listFunctions(clientConfig: ClientConfig): Promise<IPostgresFunction[]> {
```

Comment:
This should happen inside `PostgresFunctionsTreeItem.loadMoreChildrenImpl`. We want to lazy load things in the tree as much as possible, which means you don't load the functions unless they expand the "Functions" tree item